pkg/system/phase4_configuring.go

```diff
@@ -579,7 +579,17 @@ func (r *Reconciler) prepareCephBackingStore() error {
 		return fmt.Errorf("Ceph object user secret %q is not ready yet", secretName)
 	}
 
-	endpoint := "http://rook-ceph-rgw-" + r.CephObjectstoreUser.Spec.Store + "." + options.Namespace + ".svc.cluster.local:80"
+	endpoint := ""
+	if r.CephObjectstoreUser.Spec.Store != "" {
+		endpoint = "http://rook-ceph-rgw-" + r.CephObjectstoreUser.Spec.Store + "." + options.Namespace + ".svc.cluster.local:80"
+
+	} else if r.NooBaa.Labels != nil && r.NooBaa.Labels["rgw-endpoint"] != "" {
+		endpoint = r.NooBaa.Labels["rgw-endpoint"]
```

Comment:
```suggestion
	if value, ok := r.NooBaa.Labels["rgw-endpoint"]; ok {
		endpoint = fmt.Sprintf("http://%s", r.NooBaa.Labels[v])
	} else {
		return fmt.Errorf("Ceph RGW endpoint address is not available")
	}
```

Reply:
@leseb I added the protocol but I prefer to bail out also If I find the label but it is an empty value. This is why I am looking at the returned value and not the ok flag